### ISSUE TITLE
NoAuthHandlerFound at /entry/add/

### ISSUE THREAD

**Issue by krerkkiat** [marcus-yeagle/TeamLegion#1]
Not sure if this is already solved or not.

This [line](https://github.com/marcus-yeagle/TeamLegion/blob/master/entry/views.py#L51) here that produces the NoAuthHandlerFound may come from boto library that is not configured correctly.

When saving form, Django is trying to save the file user uploaded to storage which is configured in [here](https://github.com/marcus-yeagle/TeamLegion/blob/master/TeamLegion1/settings.py#L170) as AWS S3. Thus, to save file the credential to access this AWS service is needed. See more information [here](https://www.caktusgroup.com/blog/2014/11/10/Using-Amazon-S3-to-store-your-Django-sites-static-and-media-files/).

**Comment by marcus-yeagle** [comment]
I am not sure what this is referring to as I have been testing the app this morning on heroku and all the functionality seems to be working. I merged all the updates to the codebase yesterday afternoon. 

Thanks 

Marcus

**Comment by krerkkiat** [comment]
Have you try creating new entry using the page on /entry/add locally? I cloned the repository down and tried that, but it seems that the error is showing up on my local machine.

If this creating new entry works fine on Heroku, it could be the case that Heroku add some environment variables that is necessary for the boto or botocore to perform its task. In this case, it may not be a bug, but just the configuration.

**Comment by marcus-yeagle** [comment]
You are correct; Heroku has our private AWS keys stored in environment variables for the Boto configuration. You will **not** be able to run the application locally without these keys unless you reconfigure the AWS_STORAGE_BUCKET_NAME in TeamLegion1/settings.py with your own personal AWS bucket information. Additionally, you will need to obtain your own set of Clarifi API keys. 

Finally with that, run 'aws configure' in the command line as well as 'clarifi config' to setup the project locally with your keys.
